server/server.go

```diff
@@ -505,8 +505,7 @@ func (a *ArgoCDServer) newGRPCServer() *grpc.Server {
 	repoCredsService := repocreds.NewServer(a.RepoClientset, db, a.enf, a.settingsMgr)
 	var loginRateLimiter func() (io.Closer, error)
 	if maxConcurrentLoginRequestsCount > 0 {
-		loginRateLimiter = session.NewLoginRateLimiter(
-			session.NewRedisStateStorage(a.ArgoCDServerOpts.RedisClient), maxConcurrentLoginRequestsCount)
+		loginRateLimiter = session.NewLoginRateLimiter(maxConcurrentLoginRequestsCount)
```

Comment:
Since we are not using redis, we need to take into account number of api-server replicas. If limit is 50 and api-server has 2 replicas then each replica should enforce 25 limit ( 50 / 2 ).

I suggest to introduce `ARGOCD_API_SERVER_REPLICAS` env variable and use it to pass information about replicas to the api-server container ( if env variable is not set then we should assume 1 by default )

We should add variable to the HA manifest: https://github.com/argoproj/argo-cd/blob/master/manifests/ha/base/overlays/argocd-repo-server-deployment.yaml#L23